test/E2ETests/E2ETests/Helpers/TestLoggerProvider.cs

```diff
@@ -14,7 +15,7 @@ internal class TestLoggerProvider : ILoggerProvider, ILogger
     {
         private readonly IMessageSink _messageSink;
         private ITestOutputHelper _currentTestOutput;
-        IList<string> _logs = new List<string>();
+        ConcurrentQueue<string> _logs = new ConcurrentQueue<string>();
```

Comment:
There seems to be a race condition between [TimerTriggerTest](https://github.com/Azure/azure-functions-dotnet-worker/blob/31f3a8b378dd2426764ece5a9dfa976d0085e6f5/test/E2ETests/E2ETests/TimerEndToEndTests.cs#L27) and this Stream trigger test based on below logs. There is a possibility that when both TimerTriggerTest and StreamTriggerTest try to add logs in the list `_logs`, it is skipping the log (as shown in below logs). Hence, changing the datatype from `list` to a [thread-safe collection](https://learn.microsoft.com/en-us/dotnet/standard/collections/thread-safe/#fine-grained-locking-and-lock-free-mechanisms). Tried using [ConcurrentBag](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentbag-1?view=net-8.0) instead of [ConcurrentQueue](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentqueue-1?view=net-8.0) but ConcurrentBag is an unordered collection and we are testing ordering of logs in other tests.

[Logs](https://azfunc.visualstudio.com/Azure%20Functions/_build/results?buildId=157122&view=ms.vss-test-web.build-test-results-tab&runId=1362892&resultId=100068&paneView=attachments)

```csharp
System.ApplicationException : Condition not reached within timeout.

[2023-11-03T03:23:08.752Z] Executing 'Functions.BlobTriggerStreamTest' (Reason='New blob detected(LogsAndContainerScan): test-trigger-stream-dotnet-isolated/2dc72e60-92d3-4050-bc56-361f43818c89.txt', Id=ab1c808e-6f8d-4453-a942-483c9b859c25)
[2023-11-03T03:23:08.753Z] Trigger Details: MessageId: 6b9536f7-4e36-4e89-9562-704154fbfc66, DequeueCount: 1, InsertedOn: 2023-11-03T03:23:08.000+00:00, BlobCreated: 2023-11-03T03:23:06.000+00:00, BlobLastModified: 2023-11-03T03:23:06.000+00:00

// AN EXPECTED LOG LINE IS SKIPPED HERE CAUSING THE CONDITION WAITING ON THAT LOG TO TIMEOUT

[2023-11-03T03:23:08.798Z] Executed 'Functions.BlobTriggerStreamTest' (Succeeded, Id=ab1c808e-6f8d-4453-a942-483c9b859c25, Duration=47ms)
[2023-11-03T03:24:00.008Z] Executing 'Functions.TimerTrigger' (Reason='Timer fired at 2023-11-03T03:24:00.0078480+00:00', Id=e0df1011-403c-4b9c-b078-8765135e1e0b)
[2023-11-03T03:24:00.009Z] TimerInfo: {"ScheduleStatus":{"Last":"2023-11-03T03:23:00.0104776+00:00","Next":"2023-11-03T03:24:00+00:00","LastUpdated":"2023-11-03T03:23:00.0104776+00:00"},"IsPastDue":false}
[2023-11-03T03:24:00.009Z] Executed 'Functions.TimerTrigger' (Succeeded, Id=e0df1011-403c-4b9c-b078-8765135e1e0b, Duration=2ms)
```